api/token.go

```diff
@@ -63,6 +65,50 @@ func (a *API) ResourceOwnerPasswordGrant(ctx context.Context, w http.ResponseWri
 	a.issueRefreshToken(user, w)
 }
 
+// AuthorizationCodeGrant implements the authorization_code grant for use with external OAuth providers
+func (a *API) AuthorizationCodeGrant(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	code := r.FormValue("code")
+	providerName := r.FormValue("provider")
+
+	if code == "" || providerName == "" {
+		sendJSON(w, 400, &OAuthError{Error: "invalid_request", Description: "Authorization code required"})
```

Comment:
consider adding some logging around which is missing. Might be worth splitting into two checks